proxy/envoy/config.go

```diff
@@ -161,7 +161,12 @@ func build(instances []*model.ServiceInstance, services []*model.Service,
 	_ = buildFaultFilters(nil, nil)
 	_ = buildMixerCluster(mesh)
 
-	return listeners, clusters.Normalize()
+	clusters = clusters.Normalize()
+	for _, cluster := range clusters {
+		insertDestinationPolicy(config, cluster)
+	}
+
```

Comment:
We have some logic to override routes, so it's easier to keep track of used clusters by including them as part of routes. For CDS, we can include clusters referenced across all rules, and only include inbound clusters as part of the local config.